crates/api/src/wasm.rs

```diff
@@ -681,12 +681,11 @@ pub unsafe extern "C" fn wasm_instance_delete(instance: *mut wasm_instance_t) {
 
 #[no_mangle]
 pub unsafe extern "C" fn wasm_instance_new(
-    store: *mut wasm_store_t,
+    _store: *mut wasm_store_t,
```

Comment:
Perhaps we should check that the passed store matches the store associated with the module and "trap" if not?

I would like to avoid unused parameters in the C API, if possible.

Reply:
Hm yeah I'm not sure what the intention of the C API here was myself. It's theoretical that you could put the module into one cache and instance-specific data into another cache, but I'm not sure why you would do that. Ideally we'd just delete the parameter here and/or [file an issue with the upstrem spec](https://github.com/WebAssembly/wasm-c-api/issues/126). 

For now seems reasonable to return a trap!